src/libsystemd-network/sd-lldp-rx.c

```diff
@@ -147,11 +147,11 @@ static int lldp_rx_add_neighbor(sd_lldp_rx *lldp_rx, sd_lldp_neighbor *n) {
         /* Then, make room for at least one new neighbor */
         lldp_rx_make_space(lldp_rx, 1);
 
-        r = hashmap_put(lldp_rx->neighbor_by_id, &n->id, n);
+        r = hashmap_ensure_put(&lldp_rx->neighbor_by_id, &lldp_neighbor_hash_ops, &n->id, n);
         if (r < 0)
                 goto finish;
 
-        r = prioq_put(lldp_rx->neighbor_by_expiry, n, &n->prioq_idx);
+        r = prioq_ensure_put(&lldp_rx->neighbor_by_expiry, &lldp_neighbor_prioq_compare_func, n, &n->prioq_idx);
```

Comment:
function names are pointers anyway, you don't have to write the `&` before function names hence to get a pointer